x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/mapreduce/DelegatingCircuitBreakerService.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ml.aggs.mapreduce;
+
+import org.elasticsearch.common.breaker.CircuitBreaker;
+import org.elasticsearch.common.breaker.CircuitBreakingException;
+import org.elasticsearch.indices.breaker.AllCircuitBreakerStats;
+import org.elasticsearch.indices.breaker.CircuitBreakerService;
+import org.elasticsearch.indices.breaker.CircuitBreakerStats;
+
+import java.util.function.Consumer;
+
+/**
+ * Special purpose circuit breaker, don't reuse it.
+ *
+ * This breaker service and breaker wraps another breaker that has a shorter lifetime than the
+ * objects using this breaker. Therefore it is possible to disconnect the wrapped breaker.
+ *
+ * This class is only used to workaround a problem in search/aggs and should eventually be
+ * removed once search/aggs fixes problems.
```

Comment:
It would be good if this linked to an issue that describes the problem in more detail.  It's a bit cryptic to just refer to "a problem" here.

Reply:
I searched the issue list for something appropriate but couldn't find an appropriate issue, its very technical, I will try to write an appropriate issue.

The problem is described in the the aggregator: https://github.com/elastic/elasticsearch/pull/83055/files/25d3552d79e923161ad7aa83b51d615bffc4b290#diff-c419c5d769dcc959c02dfaf59aacd4ee3b47d0fac1ae32d3bafeedc3c6f3268aR75

However, I don't want to duplicate my comments. Still, it is probably best to move the comment here and link it from the aggregator (+ open and link an issue).